cpp_linter/clang_tidy.py

```diff
@@ -5,7 +5,7 @@
 from typing import Tuple, Union, List, cast
 from . import GlobalParser, CLANG_TIDY_STDOUT
 
-NOTE_HEADER = re.compile(r"^(.+):(\d+):(\d+):\s(\w+):(.*)\[(.*)\]$")
+NOTE_HEADER = re.compile(r"^([^:]+):(\d+):(\d+):\s(\w+):(.*)\[([a-z\d\-]+)\]$")
```

Comment:
My main concern here with `.+` ➡️ `[^:]+` is that Linux actually allows colons (`:`) in the file path/name. Windows and MacOS don't allow colons to be used in file names/paths (probably a legacy decision due to transport protocols like `file:///`). However, we can't assume that all users projects will be cross-platform friendly.

I also have to double check the numerous names of clang-tidy diagnostics to make sure `[a-z\d\-]+` is a compatible change.

Reply:
I have reverted the start of this pattern back to `.+` for cross-platform compatibility.

I also adjusted the pattern that matches the clang-tidy diagnostic name.